src/Tools/DTO/FunctionResponse.php

```diff
@@ -93,20 +100,52 @@ public static function getJsonSchema(): array
         return [
             'type' => 'object',
             'properties' => [
-                'id' => [
+                self::KEY_ID => [
                     'type' => 'string',
                     'description' => 'The ID of the function call this is responding to.',
                 ],
-                'name' => [
+                self::KEY_NAME => [
                     'type' => 'string',
                     'description' => 'The name of the function that was called.',
                 ],
-                'response' => [
+                self::KEY_RESPONSE => [
                     'type' => ['string', 'number', 'boolean', 'object', 'array', 'null'],
                     'description' => 'The response data from the function.',
                 ],
             ],
-            'required' => ['id', 'name', 'response'],
+            'required' => [self::KEY_ID, self::KEY_NAME, self::KEY_RESPONSE],
+        ];
+    }
```

Comment:
I must have missed it in the prior PR, but this needs to be a `oneOf`, just like for `FunctionCall`:
* Either `id` or `name` is required, not both. Both _can_ be present, but required is only one of them.
* Only `response` is always required.

We also need to add corresponding validation to the constructor, like in `FunctionCall`.

Reply:
Resolved in 3eb5279